pkg/cmd/roachtest/tests/kv.go

```diff
@@ -797,7 +797,7 @@ func registerKVRangeLookups(r registry.Registry) {
 			defer close(doneWorkload)
 			cmd := "./workload init kv --splits=1000 {pgurl:1}"
 			if err = c.RunE(ctx, c.Node(nodes+1), cmd); err != nil {
-				return err
+				t.Fatalf("failed to pre-split ranges: %v", err)
```

Comment:
This doesn't change the behavior (returning an error here has the same effect as calling `t.Fatal`). The real problem is that we don't reach `close(doneInit)`. Could you update this PR to `defer close(doneInit)` at the top?